### PR TITLE
Story1

### PR DESCRIPTION
**Pull request by jennymegan**
Story 1 complete - for code review by trainer

### CODE REVIEW

File: functions.php [hunk 1]
```diff
@@ -0,0 +1,69 @@
+<?php
+/**
+ * Connects to a named database
+ *
+ * @param $dbName database The name of the database to connect to
+ *
+ * @return databaseObject The database ready to have information extracted
+ *
+ */
+
```
Comment (mporam) [inline functions.php:10]: no line break between function and doc block

**Review by mporam — CHANGES_REQUESTED** [review]
discussed removing duplicate loops and tidying code up in person.

File: functions.php [hunk 2]
```diff
@@ -27,34 +27,23 @@ function getDatabase($dbName) {
...
-            $album = $Details['album'];
-            $year = $Details['year'];
+            $vinyl['cover_art'] = $vinyl['cover_art'] ?? 'no_img.jpg';
+            $vinyl['artist_lastname'] = $vinyl['artist_lastname'] ?? '';
 
             $result .= ' 
                  <div class="collection_item">
                   <div>
-                     <img src="' . $coverArt . '" alt="Cover Art [If Available]">
+                     <img src="'. $vinyl['cover_art'].'" alt="Cover Art [If Available]">
```
Comment (mporam) [inline functions.php]: PSR says you should have spaces either side of the concatenation operator:
` 'string' . 'string'`

File: functions.php [hunk 3]
```diff
@@ -27,34 +27,23 @@ function getDatabase($dbName) {
  */
 function populateTable(array $vinylDetails): string {
     $result = '';
-    foreach ($vinylDetails as $Details) {
-        if (isset($Details['artist_firstname']) &&
-            isset($Details['album']) &&
-            isset($Details['year'])) {
+    foreach ($vinylDetails as $vinyl) {
+        if (isset($vinyl['artist_firstname']) &&
```
Comment (mporam) [inline functions.php]: PSR2 says for long if statements you should lay them out like this:
```
if (
    condition &&
    condition &&
    condition
) {
```

**Review by mporam — CHANGES_REQUESTED** [review]

File: index.php [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+     <div class="title_bar">
+         <div class="logo"><img src="logo1.png" alt="Jenny Logo"></div>
+    <h1>MY VINYL COLLECTION</h1>
+         <input class="new_item_button" type="submit" value="Add Vinyl">
+     </div>
+     <div class="container">
+    <?php
+    include ('functions.php');
+    $db = getDatabase('collection_app');
+    $query=$db->prepare('SELECT `artist_firstname`, `artist_lastname`, `album`, `year`, `cover_art` FROM `my_vinyl_collection`; ');
```
Comment (mporam) [inline index.php]: spaces either side of the `=`

File: index.php [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+    $query->execute();
+    $vinylDetails=$query->fetchAll();
```
Comment (mporam) [inline index.php]: spaces `=`

File: newEntry.php [hunk 6]
```diff
@@ -0,0 +1 @@
+<?php
```
Comment (mporam) [inline newEntry.php]: empty file? please delete

File: stylesheet.css [hunk 7]
```diff
@@ -0,0 +1,126 @@
...
+    font-size: 0.9rem;
+    padding: 20px 0;
+    margin: 0;
+}
+
+/*Nav Bar*/
+
+.title_bar {
+    display: flex;
+    justify-content:space-around;
```
Comment (mporam) [inline stylesheet.css]: space after `:`

File: stylesheet.css [hunk 8]
```diff
@@ -0,0 +1,126 @@
...
+    border-radius: 5px;
+    margin: 10px;
+}
+
+.new_item_button:hover {
+    text-shadow: 0 0 0.2px #000500;
+    transition-delay: 0.08s;
+}
+
+.logo img{
```
Comment (mporam) [inline stylesheet.css]: space before `{`

File: test/fucntions.php [hunk 9]
```diff
@@ -0,0 +1 @@
+<?php
```
Comment (mporam) [inline test/fucntions.php]: fucn you too. please delete file

File: test/index.php [hunk 10]
```diff
@@ -0,0 +1 @@
+<?php
```
Comment (mporam) [inline test/index.php]: delete file

**Review by mporam — CHANGES_REQUESTED** [review]

File: functions.php (hunk 2 above)
Comment (jennymegan) [reply functions.php]: i was worried that was what was making the image not display. but we fixed that so I will fix this.

**Review by mporam — APPROVED** [review]

**Pull request merged**